bumble/device.py

```diff
@@ -2778,84 +2778,60 @@ def on_authentication_user_confirmation_request(self, connection, code):
         pairing_config = self.pairing_config_factory(connection)
         io_capability = pairing_config.delegate.classic_io_capability
 
-        # Respond
+        # See Bluetooth spec @ Vol 3, Part C 5.2.2.6
         if io_capability == HCI_DISPLAY_YES_NO_IO_CAPABILITY:
-            if connection.peer_pairing_io_capability in (
-                HCI_DISPLAY_YES_NO_IO_CAPABILITY,
-                HCI_DISPLAY_ONLY_IO_CAPABILITY,
-            ):
+
```

Comment:
I don't think we can remove the check for the *peer* device's capabilities here. According to Table 5.7 in Vol 3, Part C 5.2.2.6, when the peer has `NoInputNoOutput`, the responder should confirm but not show the code ("Yes/No confirmation whether to pair on device B. Device B does not show the confirmation value.")
That makes sense, because if the initiator has no input and not output, there's no point showing the code on the responder, since the initiator will auto-accept without user intervention, there's nothing to compare the code against.

Reply:
I double checked and yes, let me re-add the check :+1: